source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -1224,23 +1221,59 @@ void Graph::setUiNodeInfo(UiNodePtr node, const std::string& type, const std::st
 void Graph::createNodeUIList(mx::DocumentPtr doc)
 {
     _nodesToAdd.clear();
-    const std::string EXTRA_GROUP_NAME = "extra";
-    for (mx::NodeDefPtr nodeDef : doc->getNodeDefs())
+    const std::string EXTRA_NODES = "EXTRA_NODES";
+    std::vector<std::string> ordered_groups = {
```

Comment:
Hi @beersandrew ,
I think it's still more robust to dynamically build this list, since a scan for all of the nodedefs is done.
This allows nodegroups to be added / modified in the standard or 3rd party libraries overtime without having to rebuild this code.

For example, the Houdini and Maya nodes used their own `nodegroup` identifiers so it would work if you import the definitions from those libraries.

Reply:
I agree, however I was trying to follow the description of the issue (https://github.com/AcademySoftwareFoundation/MaterialX/issues/1541) 

>  To combat this the order of the categories should be changed to better match specification documents, e.g. Source Nodes (Texture, procedural, geometric, etc.) first, then Operator Nodes (math, adjustment, compositing, conditional, channel, etc.), then organizational and nodegraph-like nodes, then Shader BSDF Nodes, Shader EDF Nodes, etc. followed by Material nodes, etc.

@lfl-eholthouser does alphabetically make sense here to you too? Does it make sense to order by some specified list of groups and include additional groups last? 

Happy to do anything here, just trying to balance the requests

Reply:
@kwokcb Although alphabetical order is an alternative worth considering, I like the "specification order" that was originally proposed for this task.

@lfl-eholthouser Does that sound right to you as well?

Reply:
I agree, I think aligning with the specification would be best, and @beersandrew I think additional groups last makes sense.